dhcpv4/ztpv4/parse_circuitid.go

```diff
@@ -47,7 +48,13 @@ func ParseCircuitID(packet *dhcpv4.DHCPv4) (*CircuitID, error) {
 
 	// As per RFC 3046 sub-Option 1 is circuit-id. Look at 2.0 section in that RFC
 	// https://tools.ietf.org/html/rfc3046
-	circuitIdStr := string(relayOptions.Options.Get(dhcpv4.AgentCircuitIDSubOption))
+	cid := relayOptions.Options.Get(dhcpv4.AgentCircuitIDSubOption)
+	// Some Vendor like Arista sends SHIFT IN character i.e. 0x000f before circuitid
+	// remove it before matching against regexps.
+	if bytes.HasPrefix(cid, []byte{0x00, 0x0f}) {
```

Comment:
you can simply do `cid = bytes.TrimPrefix(cid, []byte{0x00, 0x0f})` here, according to the documentation:
```
TrimPrefix returns s without the provided leading prefix string. If s doesn't start
with prefix, s is returned unchanged.
```